src/Shared/EscapingUtilities.cs

```diff
@@ -79,6 +59,22 @@ internal static string UnescapeAll
             StringBuilder unescapedString = StringBuilderCache.Acquire(escapedString.Length);
 
             int currentPosition = 0;
+            int escapedStringLength = escapedString.Length;
+            if (trim)
+            {
+                while (currentPosition < escapedString.Length && Char.IsWhiteSpace(escapedString[currentPosition]))
+                {
+                    currentPosition++;
+                }
+                if (currentPosition == escapedString.Length)
+                {
+                    return String.Empty;
+                }
+                while (Char.IsWhiteSpace(escapedString[escapedStringLength]))
```

Comment:
Also, line 85:

```C#
(indexOfPercent <= (escapedStringLength - 3)) &&
```

instead of:

```C#
(indexOfPercent <= (escapedString.Length - 3)) &&
```

Reply:
Nice catch, thanks! No reason to check that digits are hex if they're white space.